src/main/kotlin/blackjack/CardsCompound.kt

```diff
@@ -0,0 +1,30 @@
+package blackjack
+
+class CardsCompound private constructor(private val set: HashSet<Int>) {
+    fun addNumber(number: CardNumber): CardsCompound {
+        return CardsCompound(
+            HashSet<Int>().apply {
+                for (score in number.getScore().number) {
+                    setNewCompound(set.toList(), score)
+                }
+            }
+        )
+    }
+
+    private fun HashSet<Int>.setNewCompound(entries: List<Int>, score: Int) {
+        for (entry in entries) {
+            val cardSum = entry + score
+            if (cardSum <= BEST) add(cardSum)
+        }
+    }
+
+    val bestNumber get() = set.toList().maxOrNull() ?: BUSTED
```

Comment:
```suggestion
    val bestNumber get() = set.maxOrNull() ?: BUSTED
```
간단하게 나타낼 수 있을 것 같아요.